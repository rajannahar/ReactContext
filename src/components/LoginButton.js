import React, {useContext} from 'react'
import { AuthContext } from '../AuthContext/AuthContext'

const LoginButton = () => {
  const { login } = useContext(AuthContext)
  return(
    <button data-testid="loginButton" onClick={ login }>Login</button>
  )
}

export default LoginButton