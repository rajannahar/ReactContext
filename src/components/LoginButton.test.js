import React from 'react';
import { render } from '@testing-library/react'
import { getByTestId, getByText } from '@testing-library/dom'
import LoginButton from './LoginButton';

test('should render login button', () => {
  const { getByTestId } = render(<LoginButton />)
  expect(getByTestId('loginButton')).toBeTruthy()
})

test('should have text "Login"', () => {
  const { getByText } = render(<LoginButton />)
  const button = getByText('Login')
  expect(button.innerHTML).toEqual('Login')
})

it("matches snapshot", () => {
  const { asFragment } = render(<LoginButton />);
  expect(asFragment()).toMatchSnapshot();
});