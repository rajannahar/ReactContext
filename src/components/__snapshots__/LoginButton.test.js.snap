// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    data-testid="loginButton"
  >
    Login
  </button>
</DocumentFragment>
`;
